sxcs.c

```diff
@@ -345,8 +294,22 @@ opt_parse(int argc, const char *argv[])
 			ret.quit_on_keypress = 1;
 		else if (strcmp(argv[i], "--mag-none") == 0)
 			ret.no_mag = 1;
-		else if (strcmp(argv[i], "--mag-filters") == 0)
-			filter_parse(argv[++i]);
+		else if (strcmp(argv[i], "--grid") == 0)
+			if (++i == argc)
+				error(1, 0, "missing argument for `%s`.", argv[--i]);
+			else
+				filter.grid = strcmp(argv[i], "no") == 0 ? NULL : grid;
+		else if (strcmp(argv[i], "--border") == 0)
+			if (++i == argc)
+				error(1, 0, "missing argument for `%s`.", argv[--i]);
+			else
+				filter.border = strcmp(argv[i], "square") == 0 ? square_border :
+						strcmp(argv[i], "circle") == 0 ? circle : NULL;
```

Comment:
maybe use a function here instead? I didn't think it necessary, Idk if there's enough filters to bother